obspy/core/util/misc.py

```diff
@@ -490,6 +490,54 @@ def factorize_int(x):
     return factors
 
 
+def get_window_times(starttime, endtime, window_length, step, offset,
+                     include_partial_windows):
+    """
+    Function calculating a list of times making up equal length windows from
+    within a given time interval.
+
+    :param starttime: The start time of the whole time interval.
+    :type starttime: :class:`~obspy.core.utcdatetime.UTCDateTime`
+    :param endtime: The end time of the whole time interval.
+    :type endtime: :class:`~obspy.core.utcdatetime.UTCDateTime`
+    :param window_length: The length of each window in seconds.
+    :type window_length: float
+    :param step: The step between the start times of two successive
+        windows in seconds. Can be negative if an offset is given.
+    :type step: float
+    :param offset: The offset of the first window in seconds relative to
+        the start time of the whole interval.
+    :type offset: float
+    :param include_partial_windows: Determines if windows that are
+        shorter then 99.9 % of the desired length are returned.
+    :type include_partial_windows: bool
+    """
+    if step > 0:
+        end = endtime.timestamp - 0.001 * step
```

Comment:
The endtime is only fed into the `np.arange` function call so if we use the precision value of the datetime objects here we might end up with numerical precision issues on the numpy side. I think using 0.1 % is pretty save.